src/inmanta/data/__init__.py

```diff
@@ -2666,8 +2666,8 @@ def _get_released_resources_base_query(
                         THEN 'orphaned'
                     ELSE r.status::text END) as status
                 FROM {cls.table_name()} as r
-                    JOIN public.configurationmodel as cm ON r.model=cm.version
-                    WHERE cm.released=TRUE
+                    JOIN public.configurationmodel as cm ON r.model=cm.version AND r.environment=cm.environment
+                    WHERE cm.released=TRUE AND r.{subquery_filter_statement}
```

Comment:
here you assume that the `subquery_filter_statement` will not be compound 

When making implicit assumptions about the query, I don't see the value of factoring it out. 

Why it can't simply be written out here (and before) as `r.environment = ?{offset}` which is a lot easier to understand?